test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ODataDynamicPropertyTests.cs

```diff
@@ -0,0 +1,868 @@
+                this.actorEntityTypeName,
+                this.awardEntityTypeName
+            };
+
+            this.context.ResolveName = type => {
+                // Lazy approach to resolving server type names - alternative would be multiple if/else blocks
+                return serverTypeNames.FirstOrDefault(d => d.EndsWith(type.Name, StringComparison.Ordinal));
+            };
+
+            var serverTypeNameToClientTypeMapping = new Dictionary<string, Type>
+            {
+                { this.genreEnumTypeName, typeof(Genre) },
+                { this.addressComplexTypeName, typeof(Address) },
+                { this.nextOfKinComplexTypeName, typeof(NextOfKin) },
+                { this.directorEntityTypeName, typeof(Director) },
+                { this.producerEntityTypeName, typeof(Producer) },
+                { this.editorEntityTypeName, typeof(Editor) },
+                { this.actorEntityTypeName, typeof(Actor) },
+                { this.awardEntityTypeName, typeof(Award) }
+            };
+
+            this.context.ResolveType = name => {
+                if (!serverTypeNameToClientTypeMapping.ContainsKey(name))
+                {
+                    return null;
+                }
+                // Lazy approach to resolving client types - alternative would be multiple if/else blocks
+                return serverTypeNameToClientTypeMapping[name];
+            };
+
+            this.typeNameToEntitySetMapping = new Dictionary<string, string>
+            {
+                { typeof(Director).Name, this.directorEntitySetName },
+                { typeof(Editor).Name, this.editorEntitySetName },
+                { typeof(Producer).Name, this.producerEntitySetName },
+                { typeof(Actor).Name, this.actorEntitySetName }
+            };
+        }
+
+        private string SerializeEntity<T>(string entitySetName, T entityObject)
+        {
+            this.context.AddObject(entitySetName, entityObject);
+            var requestInfo = new RequestInfo(this.context);
+            var serializer = new Serializer(requestInfo);
+            var headers = new HeaderCollection();
+            var entityDescriptor = new EntityDescriptor(this.clientEdmModel);
+            entityDescriptor.State = EntityStates.Added;
+            entityDescriptor.Entity = entityObject;
+            var requestMessageArgs = new BuildingRequestEventArgs("POST", new Uri("http://tempuri.org"), headers, entityDescriptor, HttpStack.Auto);
+            var odataRequestMessageWrapper = ODataRequestMessageWrapper.CreateRequestMessageWrapper(requestMessageArgs, requestInfo);
+            var linkDescriptors = new LinkDescriptor[0];
+
+            serializer.WriteEntry(entityDescriptor, linkDescriptors, odataRequestMessageWrapper);
+
+            var stream = (MemoryStream)odataRequestMessageWrapper.CachedRequestStream.Stream;
+            var streamReader = new StreamReader(stream);
+            var body = streamReader.ReadToEnd();

```

Comment:
The coding style in this project prefers ```Type```s over ```var```s

Reply:
@g2mula I'll be the first to admit that I prefer `Type`s over `var`s in a codebase with multiple contributors. I considered this but I saw that this particular project had a mix of both `var`s and `Type`s, e.g. [this](https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/ClientSerializerTests.cs) file and [this](https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/DateTimeClientSupportTest.cs) other one in the same project. My observation is that code styles are not strictly being applied to test projects. I'd say it was more of a coin flip that a deliberate choice to use `var`s here